app/models/katello/concerns/host_managed_extensions.rb

```diff
@@ -75,7 +75,18 @@ def content_and_puppet_match?
       end
 
       def inherited_attributes_with_katello
-        self.content_facet.kickstart_repository_id ||= hostgroup.inherited_kickstart_repository_id if content_facet.present?
+        if hostgroup.present?
+          if content_facet.present?
```

Comment:
We should probably check to see if the hostgroup has a content_view and lifecycle_env first.  Content facets aren't valid if both of those are not present.